src/schematools/contrib/django/migrations/0011_auto_20210623_1135.py

```diff
@@ -0,0 +1,51 @@
+# Generated by Django 3.1.12 on 2021-06-23 11:35
+
+import django.utils.timezone
+from django.contrib.postgres.fields import JSONField
+from django.db import migrations, models
+from django.db.models import F
+from django.db.models.expressions import RawSQL
+from django.db.models.functions import Cast
+
+import schematools.contrib.django.validators
+
+
+def set_defaults(apps, schema_editor):
+    """Set non nullable field `path` to value of `id`.
+    Has to use RawSql beqause F() does not support JSONfield.
+    """
+    Dataset = apps.get_model("datasets", "Dataset")
+    Dataset.objects.all().update(path=RawSQL("schema_data ->> 'id'", ()))
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("datasets", "0010_use_native_jsonfield"),
+    ]
+
+    operations = [
+        migrations.RemoveField(
+            model_name="dataset",
+            name="url_prefix",
+        ),
+        migrations.AddField(
+            model_name="dataset",
+            name="path",
+            field=models.CharField(
+                null=True,
+                max_length=250,
```

Comment:
`path` is defined to be of type `string` with format `uri-reference` right? 

Can't seem to find the definition in `amsterdam-schema` now , but I did not recall a `maxLength` to be defined on that type. Hence there should **not** be one here either. I would suggest changing the field to `TextField` and **not** specifying a `max_length`

Reply:
its no longer defined in amsterdam-schema, but equal to the path in the objectstore. This is a relic from the earlier version where it was defined in the schema. Forgot to change it. Good catch